src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -1850,15 +1890,19 @@ void ConnectivityManagerImpl::_OnWpaInterfaceScanDone(GObject * source_object, G
     g_strfreev(oldBsss);
 }
 
-CHIP_ERROR ConnectivityManagerImpl::_StartWiFiManagement(ConnectivityManagerImpl * self)
+CHIP_ERROR ConnectivityManagerImpl::_StartWiFiManagement()
 {
     // When creating D-Bus proxy object, the thread default context must be initialized. Otherwise,
     // all D-Bus signals will be delivered to the GLib global default main context.
     VerifyOrDie(g_main_context_get_thread_default() != nullptr);
 
     ChipLogProgress(DeviceLayer, "wpa_supplicant: Start WiFi management");
-    wpa_fi_w1_wpa_supplicant1_proxy_new_for_bus(G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, kWpaSupplicantServiceName,
-                                                kWpaSupplicantObjectPath, nullptr, self->_OnWpaProxyReady, nullptr);
+    wpa_fi_w1_wpa_supplicant1_proxy_new_for_bus(
+        G_BUS_TYPE_SYSTEM, G_DBUS_PROXY_FLAGS_NONE, kWpaSupplicantServiceName, kWpaSupplicantObjectPath, nullptr,
+        reinterpret_cast<GAsyncReadyCallback>(+[](GObject * sourceObject_, GAsyncResult * res_, ConnectivityManagerImpl * self) {
```

Comment:
Aren't `static_cast`s sufficient here?

Reply:
No, you can not convert different types of function signatures with `static_cast`. The signature for `GAsyncReadyCallback` is `void (*)(_GObject*, _GAsyncResult*, void*)`. So, either I can define lambda as `void *` for 3rt parameter and cast it later in the lambda body, or use proper type for user argument and cast the lambda function itself.